src/emulator/main.cpp

```diff
@@ -75,6 +75,8 @@ int main(int argc, char *argv[]) {
         return SDLInitFailed;
     }
 
+    SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 3);
+    SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 1);
```

Comment:
Unfortunately for us, it actually isn't as easy as this for macOS. Above OpenGL 2.1, macOS only supports "Core" contexts. You also need:

`SDL_GL_SetAttribute(SDL_GL_CONTEXT_PROFILE_MASK, SDL_GL_CONTEXT_PROFILE_CORE);`

Using a Core context removes all the deprecated OpenGL functionality, like vertex attrib pointers that point into RAM rather than index into a buffer object. So to upgrade OpenGL we will need to start managing vertex buffers ourselves.

Reply:
Confirmed. Using the above code, `GL_VERSION` is still `2.1` on macOS.